### PR TITLE
Backend method not found when SRT Runtime is used

### PR DESCRIPTION
**Pull request by ByronHsu**
## Motivation

```python
import sglang as sgl
from sglang.utils import http_request

model_path = "meta-llama/Meta-Llama-3.1-8B-Instruct"

if __name__ == "__main__":
    runtime = sgl.Runtime(model_path=model_path)
    sgl.set_default_backend(runtime)

    sgl.flush_cache()

    runtime.shutdown()

```

```
=== Result ===

Traceback (most recent call last):
  File "/home/jobuser/sglang/python/sglang/local_test.py", line 19, in <module>
    sgl.flush_cache()
  File "/home/jobuser/sglang/python/sglang/api.py", line 51, in flush_cache
    return backend.flush_cache()
AttributeError: 'Runtime' object has no attribute 'flush_cache'
```

The above program fails on master because `sgl.flush_cache()` is trying to find `flush_cache()` on the runtime object, but it only exists on `runtime.endpoint`. Currently, we explicitly assign `backend = runtime.endpoint` when in `run_(batch)_program`, so the program can run well. However, if we want to interact with the backend using sgl api, it will fail because runtime is not a backend. The same issue happens for `sgl.get_server_args()`

## Modifications

```python
def set_default_backend(backend: Union[BaseBackend, Runtime]):
    if isinstance(backend, Runtime):
        backend = backend.endpoint

    global_config.default_backend = backend
```

When setting the default backend, assign backend as `runtime.endpoint` directly if `backend` is an instance of `Runtime`

## Successful Test after

```python
import sglang as sgl
from sglang.utils import http_request

model_path = "meta-llama/Meta-Llama-3.1-8B-Instruct"

if __name__ == "__main__":
    runtime = sgl.Runtime(model_path=model_path)
    sgl.set_default_backend(runtime)

    sgl.flush_cache()
    print(sgl.get_server_args())

    runtime.shutdown()
```

Run successfully
```
Loading safetensors checkpoint shards:   0% Completed | 0/4 [00:00<?, ?it/s]
Loading safetensors checkpoint shards:  25% Completed | 1/4 [00:00<00:01,  1.85it/s]
Loading safetensors checkpoint shards:  50% Completed | 2/4 [00:01<00:01,  1.72it/s]
Loading safetensors checkpoint shards:  75% Completed | 3/4 [00:01<00:00,  2.48it/s]
Loading safetensors checkpoint shards: 100% Completed | 4/4 [00:01<00:00,  2.24it/s]
Loading safetensors checkpoint shards: 100% Completed | 4/4 [00:01<00:00,  2.16it/s]

{'model_path': '/shared/public/elr-models/meta-llama/Meta-Llama-3.1-8B-Instruct/07eb05b21d191a58c577b4a45982fe0c049d0693/', 'tokenizer_path': '/shared/public/elr-models/meta-llama/Meta-Llama-3.1-8B-Instruct/07eb05b21d191a58c577b4a45982fe0c049d0693/', 'tokenizer_mode': 'auto', 'skip_tokenizer_init': False, 'load_format': 'auto', 'dtype': 'auto', 'kv_cache_dtype': 'auto', 'trust_remote_code': True, 'context_length': None, 'quantization': None, 'served_model_name': '/shared/public/elr-models/meta-llama/Meta-Llama-3.1-8B-Instruct/07eb05b21d191a58c577b4a45982fe0c049d0693/', 'chat_template': None, 'is_embedding': False, 'host': '127.0.0.1', 'port': 10000, 'mem_fraction_static': 0.88, 'max_running_requests': None, 'max_total_tokens': None, 'chunked_prefill_size': 8192, 'max_prefill_tokens': 16384, 'schedule_policy': 'lpm', 'schedule_conservativeness': 1.0, 'tp_size': 1, 'stream_interval': 1, 'random_seed': 396799057, 'constrained_json_whitespace_pattern': None, 'log_level': 'error', 'log_level_http': None, 'log_requests': False, 'show_time_cost': False, 'api_key': None, 'file_storage_pth': 'SGLang_storage', 'dp_size': 1, 'load_balance_method': 'round_robin', 'dist_init_addr': None, 'nnodes': 1, 'node_rank': 0, 'json_model_override_args': '{}', 'attention_backend': 'flashinfer', 'sampling_backend': 'flashinfer', 'disable_flashinfer': False, 'disable_flashinfer_sampling': False, 'disable_radix_cache': False, 'disable_regex_jump_forward': False, 'disable_cuda_graph': False, 'disable_cuda_graph_padding': False, 'disable_disk_cache': False, 'disable_custom_all_reduce': False, 'disable_mla': False, 'enable_mixed_chunk': False, 'enable_torch_compile': False, 'max_torch_compile_bs': 32, 'torchao_config': '', 'enable_p2p_check': False, 'triton_attention_reduce_in_fp32': False, 'lora_paths': None, 'max_loras_per_batch': 8}
```

## Checklist

- [x] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/en/contributor_guide.md).
- [x] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/en/contributor_guide.md).
- [x] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

File: python/sglang/api.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
     SglSelect,
     SglVideo,
 )
+from sglang.srt.server import Runtime as SRTRuntime
```
Comment (merrymercy) [inline python/sglang/api.py]: Revert this change. See the comments in L37.
This import will introduce many dependencies (e.g., torch) that we do not want to run on the client side.

File: python/sglang/api.py [hunk 2]
```diff
@@ -35,12 +36,14 @@ def decorator(func):
...
-    from sglang.srt.server import Runtime
 
-    return Runtime(*args, **kwargs)
+    return SRTRuntime(*args, **kwargs)
 
 
-def set_default_backend(backend: BaseBackend):
+def set_default_backend(backend: Union[BaseBackend, SRTRuntime]):
+    if isinstance(backend, SRTRuntime):
+        backend = backend.endpoint
```
Comment (merrymercy) [inline python/sglang/api.py]: This change only works for `set_default_backend`.

We have some other use cases where we do not call `set_default_backend`, but directly use things like
`program.run(backend=backend)`

**Review by merrymercy — COMMENTED** [review]
Can we try to add
```
    if hasattr(backend, "endpoint"):
        backend = backend.endpoint
```
to `flush_cache` and `get_server_args` in `api.py`?

**Comment by ByronHsu** [comment]
Thanks for the review! Just made the change.
